rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize.cpp

```diff
@@ -54,10 +54,114 @@ void ResizeNode::create_node()
+    if(width_status != 0 || height_status != 0)
+        WRN("ERROR: vxCopyArrayRange _dst_roi_width or _dst_roi_height failed " + TOSTR(width_status) + "  " + TOSTR(height_status));
+    _outputs[0]->update_image_roi(_dst_roi_width_vec, _dst_roi_height_vec);
+    _dst_roi_width_vec.clear();
+    _dst_roi_height_vec.clear();
 }
 
-void ResizeNode::update_node()
-{
-  
-}
\ No newline at end of file
+void ResizeNode::init(unsigned dest_width, unsigned dest_height, RocalResizeScalingMode scaling_mode,
+                      std::vector<unsigned> max_size, RocalResizeInterpolationType interpolation_type) {
+    _scaling_mode = scaling_mode;
+    _dest_width = dest_width;
+    _dest_height = dest_height;
+    _interpolation_type = (int)interpolation_type;
+    _src_roi_size.resize(2);
+    _dst_roi_size.resize(2);
+    _max_roi_size = max_size;
+}
+
+void ResizeNode::adjust_out_roi_size() {
+    const unsigned dim = 2; // Currently supports only 2D images
+    std::vector<double> scale(dim, 1);
+    std::vector<bool> has_size(dim, false);
+    unsigned sizes_provided = 0;
+    bool has_max_size = _max_roi_size.size() > 0;
+    for (unsigned i = 0; i < dim; i++) {
+        has_size[i] = (_src_roi_size[i] != 0) && (_dst_roi_size[i] != 0);
+        sizes_provided += has_size[i];
+        scale[i] = _src_roi_size[i] ? (_dst_roi_size[i] / static_cast<double>(_src_roi_size[i])) : 1;
+    }
+    if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_STRETCH) {
+        if (sizes_provided < dim) {
+            for (unsigned i = 0; i < dim; i++) {
+                if (!has_size[i])
+                    _dst_roi_size[i] = _src_roi_size[i];
+            }
+        }
+        if (has_max_size) {
+            for (unsigned i = 0; i < dim; i++) {
+                if ((_max_roi_size[i] > 0) && (_dst_roi_size[i] > _max_roi_size[i]))
+                    _dst_roi_size[i] = _max_roi_size[i];
+            }
+        }
+    } else if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_DEFAULT) {
+        if (sizes_provided < dim) {
+            double average_scale = 1;
+            for (unsigned i = 0; i < dim; i++) {
+                if (has_size[i])
+                    average_scale *= scale[i];
+            }
+            if (sizes_provided > 1)
+                average_scale = std::pow(average_scale, 1.0 / sizes_provided);
+            for(unsigned i = 0; i < dim; i++) {
+                if(!has_size[i])
+                    _dst_roi_size[i] = std::round(_src_roi_size[i] * average_scale);
+            }
+        }
+        if (has_max_size) {
+            for (unsigned i = 0; i < dim; i++) {
+                if ((_max_roi_size[i] > 0) && (_dst_roi_size[i] > _max_roi_size[i]))
+                    _dst_roi_size[i] = _max_roi_size[i];
+            }
+        }
+    } else {
+        double final_scale = 0;
+        bool first = true;
+        for (unsigned i = 0; i < dim; i++) {
```

Comment:
this logic is very ambiguous. Please use std::min and std::max of 2 scale values instead